app/components/formatted_date/index.test.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {renderWithIntl} from '@test/intl-test-helper';
+
+import locales from '../../i18n/languages';
+
+import FormattedDate, {type FormattedDateFormat} from './index';
+
+const DATE = new Date('2024-10-26T10:01:04.653Z');
+const FORMATS = [
+    undefined,
+    {weekday: 'long'},
+    {dateStyle: 'medium'},
+    {month: 'short', day: 'numeric'},
+    {
+        month: 'short',
+        day: 'numeric',
+        hour: 'numeric',
+        minute: 'numeric',
+    },
+    {
+        year: 'numeric',
+        month: 'short',
+        day: 'numeric',
+        hour: 'numeric',
+        minute: 'numeric',
+    },
+] satisfies Array<FormattedDateFormat | undefined>;
+
+const TEST_MATRIX = Object.keys(locales).
+    map((locale) => FORMATS.map<[string, FormattedDateFormat | undefined]>((format) => [locale, format])).
+    flat(1);
+
+describe('<FormattedDate/>', () => {
+    it.each(TEST_MATRIX)("should match snapshot for '%s' locale and '%p' format", (locale, format) => {
+        const wrapper = renderWithIntl(
+            <FormattedDate
+                format={format}
+                value={DATE}
+                timezone='UTC'
+            />,
+            {locale},
+        );
+        expect(wrapper.toJSON()).toMatchSnapshot();
+    });
+});
```

Comment:
Would be nice to check timezone being passed is an object with automaticTimezone vs manualTimezone.  But that part of code is not part of your changes, so only do it if you're curious how they would impact.

Quite honestly, I'm curious how it would impact.

Reply:
Hum automaticTimezone could be problematic as it will change snapshot result with each developer TZ.

That's why I fixed it in the first place because I'm on Europe/Paris TZ while the CI server is using UTC.

But I can rewrite the test to at least use a manualTimezone and see why type checking didn't catch that ! Thanks for the suggestion !

Reply:
@KuSh You can keep the existing test and create another test case that has the use of automatic and/or manual timezone.  You're welcome to test with all the languages, but in reality, we want to make sure if anything that passes automatic or manual time zone will not cause the component to break.